webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -2480,19 +2480,101 @@ var CURRENT_GROUP_DATA={};
                     _this.toggleMediaOptions(this);
 
                 });
-                $mck_text_box.keyup(function (){
+                 _this.cursorPosition = function (element) {
+                    var selection = window.getSelection();
+                    var currentRange = selection.getRangeAt(element);
+                    var endOffset = currentRange.startOffset;
+                    var startNode = element.firstChild;
+                    var normalSpan = document.getElementById('mck-normal-span');
+                    var currentNode = selection.focusNode;
+                    if(normalSpan) {
+                        startNode = normalSpan.firstChild;
+                    }
+                    var newRange = document.createRange();
+                    newRange.setStart(startNode,0);
+                    newRange.setEnd(currentNode,endOffset);
+                    selection.removeAllRanges();
+                    selection.addRange(newRange);
+                    var position = selection.toString().length;
+                    selection.collapseToEnd();
+                    return position;
+                }
+                _this.disableSendButton = function () {
+                    var sendButton = document.getElementById('mck-msg-sbmt');
+                    sendButton.setAttribute("disabled", true);
+                    CURRENT_GROUP_DATA.DISABLE_SEND_MESSAGE = true;
+                }
+                _this.enableSendButton = function () {
```

Comment:
@hardikparashar combine disableSendButton and enableSendButton into a single function, its the same code with just a boolean difference